src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/InformAgreedExtensionDateForSpecCallbackHandler.java

```diff
@@ -65,18 +68,22 @@ protected Map<String, Callback> callbacks() {
         return Map.of(
             callbackKey(ABOUT_TO_START), this::populateIsRespondent1Flag,
             callbackKey(MID, "extension-date"), this::validateExtensionDate,
+            callbackKey(MID, "response-deadline-extension"), this::validateProposedDeadlineAdmin,
             callbackKey(ABOUT_TO_SUBMIT), this::setResponseDeadline,
             callbackKey(SUBMITTED), this::buildConfirmation
         );
     }
 
     @Override
     public List<CaseEvent> handledEvents() {
+        if (toggleService.isPinInPostEnabled()) {
```

Comment:
Look at whether we need to add any more unit/integration testing for this new functionality

Reply:
Added test. Thank you Paul!